app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/EnterMovementReferenceNumberController.scala

```diff
@@ -285,70 +285,100 @@ class EnterMovementReferenceNumberController @Inject() (
                   isAmend = true
                 )
               ),
-            movementReferenceNumber =>
-              movementReferenceNumber.value match {
+            mrnOrEntryNumber =>
+              mrnOrEntryNumber.value match {
+
                 case Left(entryNumber) =>
-                  val updatedAnswers = answers.fold(
-                    _ =>
-                      CompleteMovementReferenceNumberAnswer(
-                        Left(entryNumber)
-                      ),
-                    complete => complete.copy(movementReferenceNumber = Left(entryNumber))
-                  )
-                  val newDraftClaim  =
-                    DraftC285Claim.newDraftC285Claim.copy(movementReferenceNumberAnswer = Some(updatedAnswers))
+                  val entryNumbersMatch = fillingOutClaim.draftClaim.movementReferenceNumber
```

Comment:
entryNumbersMatch gonna be true by default (line 299)
And then you update the cache when entryNumbersMatch ??? That is exactly when you don't want to update the cache.
Rename this or rewrite

Reply:
Yep, have changed the names here with my last push